test/cmdlineTests.sh

```diff
@@ -302,25 +320,25 @@ test_solc_behaviour "${0}" "ctx:=/some/remapping/target" "" "" 1 "" "Invalid rem
 printTask "Running general commandline tests..."
 (
     cd "$REPO_ROOT"/test/cmdlineTests/
-    for tdir in ${*:-*/}
+    (( ${#selected_tests[@]} == 0 )) || selected_tests=(*)
```

Comment:
What's the purpose of this? If there are no selected tests, select all tests?
Shouldn't it be ``(( ${#selected_tests[@]} == 0 )) && selected_tests=(*)`` or ``(( ${#selected_tests[@]} > 0 )) || selected_tests=(*)`` for that?

Reply:
Ah, yeah it should be `&&` instead of `||`.

Reply:
> What's the purpose of this? If there are no selected tests, select all tests?

Yes. This is the original behavior of `cmdlineTests.sh` if there are no arguments.

> Shouldn't it be `(( ${#selected_tests[@]} == 0 )) && selected_tests=(*)` or `(( ${#selected_tests[@]} > 0 )) || selected_tests=(*)` for that?

Good catch! My original suggestion (https://github.com/ethereum/solidity/pull/11873#discussion_r702162319) was for `${#selected_tests[@]} >= 1` and then I did not notice in the review that it was different.

Reply:
solved in #11933